### DISCUSSION TITLE
Calibrate Phidget attached Thermometer

### DISCUSSION THREAD

**Discussion by techmaster70** [artisan-roaster-scope/artisan#809] [Q&A] [answered]
I have a Phidget 1048-0 and 2 Thermometers (thermocouples) attached. I can see they are off by 4-8 degrees F, but I can't figure how to 'adjust' or calibrate them. Phidget tells me it's via the software displaying the temp (so, Artisan) and send me some Python code examples.
Can anyone help with this?
I'm slightly decent at Ansible, and been exposed to Python, but really would rather find an interface setting to adjust via gui since I'm not a programmer and really have no idea where to start here. (I searched this forum but found nothing about this).

TIA
I have an Artisan 3-e roaster and Mac PowerBook Pro with OS 10.15.7.

**Comment by bigl00z3** [comment, marked as answer]
In Artisan, when you go to "Config/Device...", you go in the "Symbol ET/BT" tab
![image](https://user-images.githubusercontent.com/8886922/152415648-36e676ac-6a9c-4a4c-bebc-4e46b3b616bc.png)
In the ET & BT fields you can type a formula to compensate and correct to the real value with a formula as shown in my fake example.

**Comment by MAKOMO** [comment]
This is an FAQ. Did you study the Artisan Help page and its list of FAQ? There you find

[There is a delta between the Artisan readings and those displayed on my machine. How to fix this?](https://artisan-scope.org/help/#FAQ-Setup-9)

which links to

[Temperature Offset](https://artisan-roasterscope.blogspot.com/2019/11/symbolic-formulas-basics-new-variables.html#applications)

**Reply by techmaster70** [reply]
I did follow that... but it did not change the BT / ET readout on my screen (I have the thermometer in a glass of ice water. 2 other Thermoworks probes are reading 32 F, but the ET and BT are both off, by different amounts, 6-8 degrees. As I read it, this affects the curve in the graph... I must have misunderstood. but then why don't my values 'adjust' on the far right of the screen in the actual readout (current temp, BT, ET)? I roast by temp sometimes... like to 145 F for a Full City+, and if this is off by 8 degrees, I'm having to do the math in my head, which seems silly with all this tech around me! I'll revisit this and try again... though I do not understand the 'formula'... I'll play with it. thanks!

**Reply by MAKOMO** [reply]
Why testing in ice water? This is a temperaturepoint far away from the expected roasting temperatures. As any offset can be expected to be somewhat non-linear over the temperature range, wouldn't it makes more sense to test at boiling temperature or higher?

The expected temperature offset for standard quality TCs is 1-3.6°F.

Without you sharing what you do it is impossible to help you out here.

A formula you might put in the BT y(x) field would be "x+3" for the probe that reads 3 degrees to low to lift it up by those 3 degrees.

**Comment by poundy** [comment]
"I can see they are off by 4-8 degrees F"
Can you really, though? 
I personally think people pay way too much attention to this. Unless you have measurement noise also in that "wrong" value, recording those temperatures that the probe and module report is completely acceptable, and is just giving you a new baseline of temperature ranges that are perfectly accurate within the measurement loop you have. The one place I think it might be worth using a calculation to adjust those is if you're also recording paper notes from an on-machine display and trying to get consistency between those.  
Just remember that to be accurate in your adjustment you need to assess the range of temperatures to ensure you are not making this worse at either end of the roasting scale

**Comment by techmaster70** [comment]
That makes sense.

Much easier to test with ice water… so I started there.

I appreciated the explanation of the ‘x+3’… I assumed that was correct, but had no idea why folks were putting x in there, or why someone would put ‘x*1.03’, etc… not sure where the ‘legend’ is that explains the formula elements.

I am simply trying to get my temp readout to be exactly accurate, like I do with my smoking of meat.

Roast temp at 145… I want to know it is exactly 145.

If I can not expect that, I just need to know what I can expect.

That is what I’m trying to do.

 

Thank you. I appreciate the replies!

 

Process > Outcome

 

Chad Carrington

515-306-4090

***@***.***

 

 

From: Marko Luther ***@***.***>
Reply-To: artisan-roaster-scope/artisan ***@***.***>
Date: Friday, February 4, 2022 at 7:18 AM
To: artisan-roaster-scope/artisan ***@***.***>
Cc: techmaster70 ***@***.***>, Author ***@***.***>
Subject: Re: [artisan-roaster-scope/artisan] Calibrate Phidget attached Thermometer (Discussion #809)

 

Why testing in ice water? This is a temperaturepoint far away from the expected roasting temperatures. As any offset can be expected to be somewhat non-linear over the temperature range, wouldn't it makes more sense to test at boiling temperature or higher?

The expected temperature offset for standard quality TCs is 1-3.6°F.

Without you sharing what you do it is impossible to help you out here.

A formula you might put in the BT y(x) field would be "x+3" for the probe that reads 3 degrees to low to lift it up by those 3 degrees.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by CarefreeBuzzBuzz** [comment]
As was mentioned the offset is not linear so no matter whether you use an addition formula or multiplication formula it will very throughout the roast. I originally had that implemented in eventually abandoned it learning to trust my senses how the Roaster was behaving. 3° doesn’t matter once you become accustomed to how your Roaster works

Michael Herbert 
CarefreeBuzzBuzz
